workspaces/azure-storage-explorer/plugins/azure-storage/src/api/AzureStorageClient.ts

```diff
@@ -0,0 +1,107 @@
+import { DiscoveryApi, IdentityApi } from '@backstage/core-plugin-api';
+import { ResponseError } from '@backstage/errors';
+import { AzureStorageApi } from './AzureStorageApi';
+
+export class AzureStorageClient implements AzureStorageApi {
+  private readonly discoveryApi: DiscoveryApi;
+  private readonly identityApi: IdentityApi;
+
+  public constructor(options: {
+    discoveryApi: DiscoveryApi;
+    identityApi: IdentityApi;
+  }) {
+    this.discoveryApi = options.discoveryApi;
+    this.identityApi = options.identityApi;
+  }
+
+  public async listStorageAccounts(): Promise<any> {
+    const baseUrl = `${await this.discoveryApi.getBaseUrl('azurestorage')}`;
+    const listContainersUrl = `${baseUrl}/list/accounts`;
+    const { token: idToken } = await this.identityApi.getCredentials();
+    const response = await fetch(listContainersUrl, {
```

Comment:
You want to use the `FetchApi` here is it will wrap the call with the Backstage token.

Here's how we did it for the Azure DevOps plugin: https://github.com/backstage/community-plugins/blob/d923c72223e23aa7b84c887941eb0ae64eafe0a4/workspaces/azure-devops/plugins/azure-devops/src/api/AzureDevOpsClient.ts#L229

Reply:
Updated to use `FetchApi`